lib/spawn-node.js

```diff
@@ -17,6 +18,7 @@ module.exports = function (collection, nodeName, index, cb) {
   var lookup = context.nodeInfo.lookup
   var info = lookup[nodeName]
   if (info) {
+    if (info.name === 'dat') return createDatNode(collection, index, context, fileObject, cb)
```

Comment:
Sweet hack!

I think something like this could be made official by making it a little more generic. It could be something more like this:

```js
if (typeof info.action === 'function') return info.action({collection, index, context, fileObject}, cb)
```

And then you would move createDatNode into `nodes/dat-chunk/index.js`.

```js
var createDatNode = require('./create-dat-node')
module.exports = {
  name: 'dat',
  node: 'dat-chunk',
  action: (opts, cb) => {
    createDatNode(opts.collection, opts.index, opts.context, opts.fileObject, cb)
  },
  group: 'simpleChunks',
  description: 'Load a chunk from dat p2p'
}
```

Reply:
Thanks, That's a helpful pointer. I will adjust the code it to that.